tests/integration/test_schema.py

```diff
@@ -1250,48 +1396,77 @@ async def test_schema(registry_async_client, trail):
                 "name": "first_name",
                 "type": "string",
             },
+            {
+                "name": f"{unique}",
```

Comment:
This seems odd, the unique marker shouldn't be reused, or the old schema should be used here.

Reply:
I reworked on the test, please have a look.

Maybe all the places where uniqueness is needed, the code should use the `name` of the AVRO schema to make the schema unique?